Applications/DataExplorer/DataView/MeshElementRemovalDialog.cpp

```diff
@@ -170,13 +170,13 @@ void MeshElementRemovalDialog::on_materialIDCheckBox_toggled(bool is_checked)
 	{
 		this->materialListWidget->clear();
 		_matIDIndex = _currentIndex;
-		const std::vector<MeshLib::Element*> elements (_project.getMesh(this->meshNameComboBox->currentText().toStdString())->getElements());
-		auto it = std::max_element(elements.begin(), elements.end(),
-			[](MeshLib::Element const*const x, MeshLib::Element const*const y) { return x->getValue() < y->getValue(); }
-		);
-		unsigned max_material ((*it)->getValue());
+		auto mesh = _project.getMesh(this->meshNameComboBox->currentText().toStdString());
+		const std::vector<MeshLib::Element*> elements = mesh->getElements();
 
-		for (unsigned i=0; i<=max_material; ++i)
+		auto materialIds = mesh->getProperties().getPropertyVector<int>("MaterialIDs");
```

Comment:
The problem is larger than that. MaterialIDs mustn't be selectable if no material array exists. Everything afterwards depends on that and I'd have to look closer to see where more detailed checks need to be performed. I'll fix the class on Monday morning if that's alright with everyone.